drivers/net/wireless/mediatek/mt76/mt7921/mac.c

```diff
@@ -1339,13 +1346,16 @@ void mt7921_mac_reset_work(struct work_struct *work)
 	ieee80211_iterate_active_interfaces(hw,
 					    IEEE80211_IFACE_ITER_RESUME_ALL,
 					    mt7921_vif_connect_iter, NULL);
+	clear_bit(MT76_HW_RESET, &dev->mphy.state);
+	mt76_connac_power_save_sched(&dev->mt76.phy, &dev->pm);
 }
 
 void mt7921_reset(struct mt76_dev *mdev)
 {
 	struct mt7921_dev *dev = container_of(mdev, struct mt7921_dev, mt76);
 
-	queue_work(dev->mt76.wq, &dev->reset_work);
+	if (!test_bit(MT76_HW_RESET, &dev->mphy.state))
+		queue_work(dev->mt76.wq, &dev->reset_work);
```

Comment:
@objelf do we really need this check? A similar order is already performed by queue_work() and mt76 is an ordered workqueue